### ISSUE TITLE
Bug in Ring2D initialization

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#6178]
The following should work, but doesn't:

```python
In [1]: from astropy.modeling import models

In [2]: ring = models.Ring2D(amplitude=1, x_0=1, y_0=1, r_in=2, r_out=3)
---------------------------------------------------------------------------
InputParameterError                       Traceback (most recent call last)
<ipython-input-2-6a1a8f7d27ef> in <module>()
----> 1 ring = models.Ring2D(amplitude=1, x_0=1, y_0=1, r_in=2, r_out=3)

/Users/tom/Dropbox/Code/Astropy/astropy/astropy/modeling/functional_models.py in __init__(self, amplitude, x_0, y_0, r_in, width, r_out, **kwargs)
   1374             if width is not None:
   1375                 raise InputParameterError(
-> 1376                     "Cannot specify both width and outer radius separately.")
   1377             width = r_out - r_in
   1378         elif width is None:

InputParameterError: Cannot specify both width and outer radius separately.
```

I think this is because the default width is not None.

**Comment by pllim** [comment]
This was added by @adonath 4 years ago, maybe he can comment?

**Comment by adonath** [comment]
From a quick look it seems like the behaviour was broken with the introduction of default values for the parameters. The default for width used to be `width = None`, while it is now `width = width.default = 1`, which breaks the consistency check. The solution is probably to check, whether `width == width.default` instead of  `width is None`.

**Comment by pllim** [comment]
Thanks, @adonath . Potential fix in #6192.
